platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Submission.java

```diff
@@ -2,29 +2,50 @@
 
 import javax.persistence.*;
 import java.time.ZonedDateTime;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 /**
- * A submission for a (task, challenge, user)
+ * A submission for a (task, challenge, user). This
+ * represents what the user uploaded, e.g. gave us
+ * to execute.
  */
 @Entity
 @Table(name = "submission")
-public class Submission extends IdentifiableEntity implements StoredObject {
+public class Submission extends IdentifiableEntity {
     @ManyToOne(optional = false)
     private TaskResult taskResult;
 
     @OneToMany(mappedBy = "submission")
     private List<TestResult> testResults;
 
+    /**
+     * Language this submission is in. Not that as we have
+     * multiple files bundled in one Submission, the individual
+     * files may well use different programming languages, so we
+     * might use a more sophisticated way of encoding this as soon
+     * as necessary.
+     */
     @ManyToOne
     private Language language;
 
-    @Column(name = "file_name")
-    private String fileName;
+    /**
+     * Names of the files associated with this submission.
+     * When querying storage for files, use exactly these names.
+     */
+    @Column(name = "file_names")
+    private Set<String> fileNames;
```

Comment:
Wouldn't it be nicer if we just saved the path to the directory root and not care what's inside? 
The only ones who should care(and will handle stuff accordingly) are the runtime and the browser when showing/downloading the code.